torchgen/model.py

```diff
@@ -946,6 +978,33 @@ def native_function_class_name(self) -> Optional[str]:
             # updating every kernel definition + callsite of every in-tree aten kernel.
             return None
 
+    def should_gen_dispatchless_composite(self, f: NativeFunction) -> bool:
+        if f.func.name in self.index:
+            return False
+        return len(f.composite) > 0 and self.dispatch_key in COMPOSITE_DISPATCH_KEYS
+
+    def dispatchless_composite_struct(self, f: NativeFunction) -> str:
+        name = f.func.name.unambiguous_name()
+        return f"{name}_{self.dispatch_key.lower()}"
+
+    def dispatchless_composite_kernel(self, f: NativeFunction) -> str:
+        struct_name = self.dispatchless_composite_struct(f)
+
+        # Similar to cpp.name().
+        # Couldn't use it due to circular dependency issues.
+        name = str(f.func.name.name)
+        if f.func.is_out_fn():
+            name += "_out"
+
+        return f"{name}<{struct_name}>"
+
+    def has_registered_kernel(
+        self, f: NativeFunction, g: Optional[NativeFunctionsGroup]
+    ) -> bool:
+        if g is not None and g.structured:
+            return self.has_kernel(g) or self.dispatch_key == DispatchKey.Meta
+        return self.has_kernel(f)
```

Comment:
There are a couple of problems here:
1. If `NativeFunction` is a functional overload (i.e. without CPU or CUDA dispatch), the `index` won't find a `BackendMetadata` for it. The `index` only has an entry for the out overload.
2. Since structured kernels don't explicitly have a dispatch entry for Meta, they won't be found in the `index`. I could add them when parsing the YAML file, but I didn't know whether that would be okay

Reply:
This is reasonable. We can leave this as is for now. It seems like there is a general pattern where in the codegen we want to be able to call a native function at a particular dispatch key, but we need to know if it exists, and probably we should make some auxiliary data structure to make this easier to do.